test/provider_os_memory_multiple_NUMA_nodes.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+
+#include "cpp_helpers.hpp"
+#include "provider_os_memory_internal.h"
+#include "umf/providers/provider_os_memory.h"
+#include <numa.h>
+#include <numaif.h>
+#include <umf/memory_provider.h>
+
+using umf_test::test;
+
+static umf_os_memory_provider_params_t UMF_OS_MEMORY_PROVIDER_PARAMS_TEST = {
+    /* .protection = */ UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+    /* .visibility = */ UMF_VISIBILITY_PRIVATE,
+
+    // NUMA config
+    /* .nodemask = */ NULL,
+    /* .maxnode = */ 0,
+    /* .numa_mode = */ UMF_NUMA_MODE_BIND,
+    /* .numa_flags = */ 0,
+
+    // others
+    /* .traces = */ 1,
+};
+
+std::vector<std::tuple<int, int>> get_available_NUMA_nodes_numbers() {
+    int max_node = numa_max_node() + 1;
```

Comment:
`max_node` is confusing - it's not really a max node id but rather number of nodes - please rename it.

Reply:
no Igor - this is a max node ID. Consider the system where numa nodes have IDs 0, 1 and 3. Then the max numa ID would be 3.

Reply:
Done.